src/adapter/activateDaffodilDebug.ts

```diff
@@ -209,19 +259,43 @@ export function activateDaffodilDebug(
         createDebugRunFileConfigs(resource, 'run', 'execute')
       }
     ),
-    vscode.commands.registerCommand('extension.dfdl-debug.copyTDML', (_) => {
-      // Ask for destination path
-      // Copy file in /tmp to destination path
-      // TDMLConfig.path should not be used here because that only matters when sending to the server
-      // We could make it so that if someone wants to specify the path and set the action to 'copy', but
-      //   that doesn't make a whole lot of sense.
-      let targetResource = vscode.commands.executeCommand(
-        'extension.dfdl-debug.getTDMLPath'
-      )
+    vscode.commands.registerCommand(
+      'extension.dfdl-debug.copyTDML',
+      async (_) => {
+        // Ask for destination path
+        // Copy file in /tmp to destination path
+        // TDMLConfig.path should not be used here because that only matters when sending to the server
+        // We could make it so that if someone wants to specify the path and set the action to 'copy', but
+        //   that doesn't make a whole lot of sense.
+        let targetResource = await vscode.commands.executeCommand(
+          'extension.dfdl-debug.getValidatedTDMLCopyPath'
+        )
 
-      // Is there a better way of error checking this?
-      fs.copyFileSync(getTmpTDMLFilePath(), targetResource as unknown as string)
-    }),
+        // Is there a better way of error checking this?
+        if (targetResource) {
+          copyTestCase(
+            getTmpTDMLFilePath(),
+            targetResource as unknown as string
+          )
+            .then((copiedBuffer) => {
+              fs.writeFileSync(
+                targetResource as unknown as string,
+                xmlFormat(copiedBuffer)
+              )
+            })
+            .catch((reason) => {
+              // Not sure if we need to do something different/more here
```

Comment:
Questin: What happens if `catch` is removed? Does the whole extension stop, or is the error sent to the console anyway?